### PR TITLE
Allow passing the source code to parse directly as a command line argument

### PR DESCRIPTION
**Pull request by ahoppen**
If you just want to parse a single line code snippet, you previously had to either create a file with the contents or pass the contents to stdin, neither of which was super convenient. Add a `-s`/`--source` option to pass the source text directly. 

And while doing this, I also split `swift-parser-cli` up into multiple files (first commit, no functionality change) and changed all commands from classes to structs (last commit).

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/swift-parser-cli/ParseCommand.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+/// A command  that has arguments to parse source code
+protocol ParseCommand {
+  var arguments: ParseArguments { get }
+}
+
+extension ParseCommand {
+  /// The name of the source file that can can be displayed to users
+  ///
+  /// This is not necesarrily a file path if the file contents are read from
```
Comment (bnbarham) [inline Sources/swift-parser-cli/ParseCommand.swift:34]: `necessarily`

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by kimdv — APPROVED** [review]

**Comment by ahoppen** [comment]
pt

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by kimdv** [comment]
@swift-ci Please test Windows

**Comment by kimdv** [comment]
@swift-ci please test windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**